/*
 * This file was automatically generated by EvoSuite
 * Tue Jul 31 19:03:55 GMT 2018
 */

package org.mozilla.javascript.ast;

import org.junit.Test;
import static org.junit.Assert.*;
import static org.evosuite.runtime.EvoAssertions.*;
import org.evosuite.runtime.EvoRunner;
import org.evosuite.runtime.EvoRunnerParameters;
import org.junit.runner.RunWith;
import org.mozilla.javascript.ast.AstNode;
import org.mozilla.javascript.ast.AstRoot;
import org.mozilla.javascript.ast.ContinueStatement;
import org.mozilla.javascript.ast.ErrorNode;
import org.mozilla.javascript.ast.ForLoop;
import org.mozilla.javascript.ast.Name;

@RunWith(EvoRunner.class) @EvoRunnerParameters(mockJVMNonDeterminism = true, useVFS = true, useVNET = true, resetStaticState = true, separateClassLoader = true, useJEE = true) 
public class ForLoop_ESTest extends ForLoop_ESTest_scaffolding {

  @Test(timeout = 4000)
  public void test0()  throws Throwable  {
      StringBuilder stringBuilder0 = new StringBuilder();
      ForLoop forLoop0 = new ForLoop();
      ContinueStatement continueStatement0 = new ContinueStatement(4, 0, (Name) null);
      forLoop0.setInitializer(continueStatement0);
      AstNode.DebugPrintVisitor astNode_DebugPrintVisitor0 = new AstNode.DebugPrintVisitor(stringBuilder0);
      // Undeclared exception!
      try { 
        forLoop0.visit(astNode_DebugPrintVisitor0);
        fail("Expecting exception: NullPointerException");
      
      } catch(NullPointerException e) {
         //
         // no message in exception (getMessage() returned null)
         //
         verifyException("org.mozilla.javascript.ast.ForLoop", e);
      }
  }

  @Test(timeout = 4000)
  public void test1()  throws Throwable  {
      ForLoop forLoop0 = new ForLoop((-532));
      AstRoot astRoot0 = new AstRoot();
      forLoop0.setInitializer(astRoot0);
      forLoop0.setCondition(astRoot0);
      // Undeclared exception!
      try { 
        forLoop0.toSource();
        fail("Expecting exception: NullPointerException");
      
      } catch(NullPointerException e) {
         //
         // no message in exception (getMessage() returned null)
         //
         verifyException("org.mozilla.javascript.ast.ForLoop", e);
      }
  }

  @Test(timeout = 4000)
  public void test2()  throws Throwable  {
      ForLoop forLoop0 = new ForLoop((-442));
      ErrorNode errorNode0 = new ErrorNode(19, (-2013));
      forLoop0.setIncrement(errorNode0);
      assertEquals(120, forLoop0.getType());
  }

  @Test(timeout = 4000)
  public void test3()  throws Throwable  {
      ForLoop forLoop0 = new ForLoop();
      forLoop0.getInitializer();
      assertEquals("120", forLoop0.toString());
  }

  @Test(timeout = 4000)
  public void test4()  throws Throwable  {
      ForLoop forLoop0 = new ForLoop(1701, 0);
      assertEquals(120, forLoop0.getType());
  }

  @Test(timeout = 4000)
  public void test5()  throws Throwable  {
      ForLoop forLoop0 = new ForLoop((-532));
      forLoop0.getCondition();
      assertEquals(120, forLoop0.getType());
  }

  @Test(timeout = 4000)
  public void test6()  throws Throwable  {
      ForLoop forLoop0 = new ForLoop();
      forLoop0.getIncrement();
      assertEquals("120", forLoop0.toString());
  }
}
